### PR TITLE
HTTP -> HTTPS except POM headers and Maven Wrapper files (#120)

### PR DESCRIPTION
**Pull request by LIttleAncientForestKami**
## Overview

- Java files have Javadocs changed: URLs now use https
- `grep http **/* -r | grep -v https` was used for testing / looking for places to change
- all links I changed worked in their changed version
- the issue says (in bold even) all files, so do let me know if it really is what was meant @sormuras and @marcphilipp - I didn't fully follow, since:
- Maven wrapper files are IIRC autogenerated, so I did not make changes there
- Maven POM documentation and example use http for XML namespaces in the headers, and both http and https link fail their resolution. Example docs: https://maven.apache.org/pom.html  - so XML headers I made as Maven docs have them, in case some legacy tooling needs that.

Let me know what else's needed.
---
I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: junit5-jupiter-starter-maven/pom-SNAPSHOT.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (sormuras) [inline junit5-jupiter-starter-maven/pom-SNAPSHOT.xml:3]: IIRC, this `http` usage is for the namespace reference only. Please revert it to `http`.

File: junit5-jupiter-starter-maven/pom.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (sormuras) [inline junit5-jupiter-starter-maven/pom.xml:3]: Same as above, please revert it to `http`.

**Review by sormuras — CHANGES_REQUESTED** [review]
Thanks for submitting this PR Tomasz - please address the request changes.

File: junit5-jupiter-starter-maven/pom-SNAPSHOT.xml (hunk 1 above)
Comment (sbrannen) [reply junit5-jupiter-starter-maven/pom-SNAPSHOT.xml:3]: The URL on the left is actually the "name" of the namespace.

See the `targetNampespace` attribute in the documented referenced by the URL on the right.

The URL on the right will be used by tools (or humans) to download the XSD.

Thus, `https` is correct for the URL on the right.

File: junit5-jupiter-starter-maven/pom.xml (hunk 2 above)
Comment (sbrannen) [reply junit5-jupiter-starter-maven/pom.xml:3]: See comments above about keeping `https` for the URL on the right.

**Comment by sbrannen** [comment]
Thanks for the PR.

It would be better to introduce [nohttp](https://github.com/spring-io/nohttp) into the build(s) instead of manually grepping.

Perhaps you'd like to introduce `nohttp` with this PR.

**Comment by sbrannen** [comment]
Note that `nohttp` is already used in the builds for JUnit 5. See:

- https://github.com/junit-team/junit5/blob/main/build.gradle.kts
- https://github.com/junit-team/junit5/blob/main/settings.gradle.kts

**Comment by sormuras** [comment]
I guess, I found another candidate for adding a `s` to right URI: https://github.com/junit-team/junit5-samples/blob/891ae184c2c627beefc3ddfa0758f466e45ace2b/junit5-migration-maven/pom.xml#L3

Is there a standalone `nohttp` variant, not needing a build tool integreation?
Asking for "ant", "bazel", and "modular-world" samples.

**Review by sormuras — APPROVED** [review]

**Comment by sbrannen** [comment]
> I guess, I found another candidate for adding a `s` to right URI:

`nohttp` _should_ inform us about everything this PR misses, and it will do that for every build.

> Is there a standalone nohttp variant, not needing a build tool integreation?

Yes. Check out the project page. There's a CLI, Checkstyle support, and Gradle support.

https://github.com/spring-io/nohttp#modules

**Comment by LIttleAncientForestKami** [comment]
Hi fellas! Thanks for lots of good points, will go through them soon. I
just became a father!! Writing this to not ghost you and let you know.
Nohttp sounds like just what is needed and remarks about XML headers are on
point.

Once I'm done should I squash the commits?

**Comment by sbrannen** [comment]
> I just became a father!! 

Congratulations! 👶 

> Nohttp sounds like just what is needed and remarks about XML headers are on point. 

👍 

> Once I'm done should I squash the commits?

Yes, that would be fine. Feel free to rename the issue, rework the description, squash commits, rebase on `main`, and force push the changes to your PR branch.

**Review by donphelix — APPROVED** [review]
LGTM

**Comment by LIttleAncientForestKami** [comment]
Thank you for the review and approve!

Also, where's my thumb up Jesus? 😉 (Seeing lgtm without it feels
incomplete now...
https://memegenerator.net/instance/65177257/thumbs-up-jesus-1000-lines-of-code-changed-lgtm
)

śr., 6 lis 2024, 09:25 użytkownik Don Phelix ***@***.***>
napisał:

**Comment by marcphilipp** [comment]
@LIttleAncientForestKami Thanks for your initial work! I just pushed it over the finish line. There's now a GitHub Action that runs nohttp.

**Pull request merged**